tests/e2e/vm_label_annotation_test.go

```diff
@@ -0,0 +1,251 @@
+package e2e
+
+import (
+	"fmt"
+	"io/fs"
+	"path/filepath"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/deckhouse/virtualization/tests/e2e/executor"
+	kc "github.com/deckhouse/virtualization/tests/e2e/kubectl"
+)
+
+var _ = Describe("Label and Annotation", Ordered, ContinueOnFailure, func() {
+	imageManifest := vmPath("image.yaml")
+	vmManifest := vmPath("vm_label_annotation.yaml")
+
+	const (
+		labelName       = "os"
+		labelValue      = "ubuntu"
+		annotationName  = "test-annotation"
+		annotationValue = "true"
+	)
+
+	getPodName := func(resourceName string) string {
+		getPodCMD := "get pod --no-headers -o custom-columns=':metadata.name'"
+		subCMD := fmt.Sprintf("-n %s %s | grep %s", conf.Namespace, getPodCMD, resourceName)
+		podCMD := kubectl.RawCommand(subCMD, ShortWaitDuration)
+		podName := strings.TrimSuffix(podCMD.StdOut(), "\n")
+		return podName
+	}
+
+	getRecourseLabel := func(resourceType kc.Resource, resourceName string) *executor.CMDResult {
+		label := fmt.Sprintf("jsonpath='{.metadata.labels.%s}'", labelName)
+		cmdResult := kubectl.GetResource(resourceType, resourceName, kc.GetOptions{
+			Output:    label,
+			Namespace: conf.Namespace,
+		})
+		return cmdResult
+	}
+
+	getRecourseAnnotation := func(resourceType kc.Resource, resourceName string) *executor.CMDResult {
+		annotation := fmt.Sprintf("jsonpath='{.metadata.annotations.%s}'", annotationName)
+		cmdResult := kubectl.GetResource(resourceType, resourceName, kc.GetOptions{
+			Output:    annotation,
+			Namespace: conf.Namespace,
+		})
+		return cmdResult
+	}
+
+	WaitVmStatus := func(name, phase string) {
+		GinkgoHelper()
+		WaitResource(kc.ResourceVM, name, "jsonpath={.status.phase}="+phase, LongWaitDuration)
+	}
```

Comment:
Why not create an additional helper WaitResourceStatusPhase?

Reply:
Not in this PR